### PR TITLE
Clean up README

### PR DESCRIPTION
**Pull request by dblarons**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,65 +2,113 @@
...
-into one with a more declarative syntax
+This code has a few problems:
+
+1. **Mutability makes the type hard to reuse.**
+
+   It is hard to reuse the FirstName type in another place that requires
+   slightly modified validation, e.g. a max length restriction, because
+   `getValidationErrorMessage` is set as a static function on the type.
+
+2. **Mutability makes the options object is hard to reuse.**
```
Comment (dorisip) [inline README.md]: grammar issue here

File: README.md [hunk 2]
```diff
@@ -2,65 +2,113 @@
...
 `npm install tcomb-builder tcomb-validation`
 
-## Summary
+## Benefits
+
+1. Immutability
+
+2. Replaces deeply nested JSON options objects in your codebase
+
+3. Yields a standard tcomb type and options JSON object
```
Comment (dorisip) [inline README.md]: you have `JSON options objects` above. and here you have `options JSON object`

I think you refer to it as `option object` in the rest of the doc

Comment (dblarons) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (dblarons) [reply README.md]: done

**Review by liamcain — APPROVED** [review]

**Review by dorisip — APPROVED** [review]

**Pull request merged**